src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp

```diff
@@ -133,11 +133,48 @@ memory::ptr ocl_engine::allocate_memory(const layout& layout, allocation_type ty
                     "Requested ", layout.bytes_count(), " bytes "
                     "but max alloc size is ", get_device_info().max_alloc_mem_size, " bytes");
 
-    auto used_mem = get_used_device_memory(allocation_type::usm_device) + get_used_device_memory(allocation_type::usm_host);
-    OPENVINO_ASSERT(layout.bytes_count() + used_mem <= get_max_memory_size(),
+    auto dev_type = get_device_info().dev_type;
+    auto used_usm_device = get_used_device_memory(allocation_type::usm_device);
+    auto used_usm_host = get_used_device_memory(allocation_type::usm_host);
```

Comment:
1. Could you extract those checks to separate method (can_allocate() or somethine like that)
2. Please add a comment which describes OS/device specific limitations on allocation size
3. Can the code be simplified? The error message is duplicated many times with minor modifications, so I think it can be parameterized and then saved to stringstream for instance.

Reply:
Yes, I was going to reduce the condition because we need to allow swapping for device memory too.

Reply:
Updated, could you review again?